### ISSUE TITLE
[Dev UI] Add "run eval" button to the dataset detail page

### ISSUE THREAD

**Issue by mjchristy** [firebase/genkit#1503]
## Overview

TL:DR: Add the ability to run an evaluation right from a dataset.

**User goals**

When you try to run an evaluation, you get taken to the dataset page for the first time. It would be nice to be able to close the loop and offer a way to run evals right from a dataset.

**Requirements**
- Disable the button if there is no data in the dataset
- Select the right type if possible: flow or modal
- If there is a target flow, choose that as the flow in the new eval dialog.
- Move the download button into the overflow

Mocks:
![Image](https://github.com/user-attachments/assets/478810c2-dc99-44a5-a116-39eee6a024ad)
![Image](https://github.com/user-attachments/assets/f75bc03f-edc9-4df6-baca-e5ae8af25831)

Notes:
We may revisit these buttons later, but this should be a quick solve for our dead-end problem.

**Comment by mjchristy** [comment]
As discussed in the meeting:

We want this button to redirect to the Create New Evaluation page, with the dataset filled out.
